src/modules/commander/commander.cpp

```diff
@@ -2747,6 +2740,20 @@ Commander::run()
 
 		arm_auth_update(now, params_updated || param_init_forced);
 
+		// Handle shutdown request from emergency battery action
+		if(dangerous_battery_level_requests_poweroff){
```

Comment:
There's a minimal chance that the system is armed now, so that should be handled.